content/articles/json-server-and-axios/index.md

```diff
@@ -0,0 +1,398 @@
+### Introduction
+Vue is a front-end JavaScript framework used to create single-page apps that run on the client. It can also be used to create full-stack applications by making HTTP requests to a backend server. It is popular with Node.js and Express(MEVN stack).
+
+JSON server is an npm package that lets you create mock REST APIs with zero coding. It is used to create a simple JSON file that can be used as a database and responds to HTTP requests.
+
+Axios is a HTTP client to make HTTP requests to the JSON Server.
+
+In this article, we will build a shopping list application. We will start with a blank Vue.js application then add `json-server` for local data storage, and Axios for making HTTP requests.
+
+### Table of contents
+- [New Vue.js application](#new-vuejs-application)
+- [Creating a JSON file and installing JSON-Server](#creating-a-json-file-and-installing-json-server)
+- [Getting HTTP data within our application](#getting-http-data-within-our-application)
+- [Adding data using POST](#adding-data-using-post)
+- [Updating data using PATCH](#updating-data-using-patch)
+- [Deleting data using DELETE](#deleting-data-using-delete)
+
+### Prerequisites
+To follow along with this tutorial, you are required to have the following:
+- [Node.js](https://nodejs.org/en/download/package-manager/) installed
+- Some knowledge of JavaScript
+
+### New Vue.js application
+Assuming you have installed [Node.js](https://nodejs.org/en/download/package-manager/) in your computer, run the following commands in your terminal to install Vue CLI:
+
+```bash
+#Install Vue.js CLI globally
+$ npm install -g @vue/cli
+```
+
+After the installation, create a new project using:
+
+```bash
+$ vue create shopping-list 
+```
+
+Navigate to the folder and open the project in your favorite editor. e.g VS Code:
+
+```bash
+$ cd shopping-list
+
+# open project in vs code
+$ code .
+```
+
+Run the following command to start the Vue.js application:
+
+```bash
+$ npm run serve
+```
+
+Our application will run on <http://localhost:8080/> in the browser. We should now see a default homepage of Vue running on our browser.
+
+### Creating a JSON file and installing JSON server
+We can now create a simple JSON file that can be used as a database.
+
+Inside our shopping-list directory, let's create a file named `data.json`:
+
+```bash
+$ touch data.json
+```
+
+Let's add the following data in the file:
+
+```json
+{
+  "items":[
+      {
+      "id":0,
+      "name":"Carrots"
+      },
+      {
+      "id":1,
+      "name":"Cabbage"
+      },
+      {
+      "id":2,
+      "name":"Banana"
+      },
+      {
+      "id":0,
+      "name":"Cakes"
+      }
+  ]
+}
+```
+
+From the code above, we have created a shopping list but we need to install `json-server` and start it:
+
+```bash
+#Install json-server globally
+$ npm install json-server -g 
+
+#Inside your working directory, start the JSON Server
+$ json-server data.json
```

Comment:
For certain bash code snippets, you have used `$` and for certain snippets you haven't. Make sure to maintain consistency.